docs/architecture/ADR-001-interest-stream.md

```diff
@@ -0,0 +1,118 @@
+
+As an alternative, transfer of uTokens via IBC could be forbidden or unsupported - in that case, only the scenarios where uToken balances are held on Umee and Ethereum need to be addressed.
+
+## Decision
+
+"uToken to base asset exchange rate grows over time". This method is inspired by the Compound model.
+
+- Requirement: Umee chain stores uAsset <-> Asset exchange rate for each asset (not 1:1)
+
+In this implementation, for each whitelisted Cosmos asset type, the Umee chain stores an "exchange rate" between the base asset and its associated uToken. The exchange rate starts equal to 1, and increases whenever interest would have been applied to uToken balances in the original implementation. Whenever a lender deposits or withdraws base assets for uToken, this exchange rate is used.
+
+Example scenario:
+
+> Two lenders Alice and Bob provide Atoms to the asset facility at different times and earn interest. Assume that for the duration of this scenario, the interest on deposited uAtoms is 0.1 percent per week (or 1 atom per week per 1000 deposited).
+>
+> The asset facility starts with 0 atoms in custody and 0 uAtoms in circulation. The exchange rate of Atom:uAtom starts at 1.
+>
+> At a time we will label week=0, Alice deposits 2000 atoms and receives 2000 uAtoms per the exchange rate.
+>
+> At week=1, the exchange rate Atom:uAtom increases to 1.001. While Alice's uAtom balance is 2000, it is now worth 2002 Atoms if she were to redeem it.
+>
+> At week=2, Atom:uAtom increases to about 1.002 (technically 1.002001, but approximate amount will be displayed here)
+>
+> Also at week=2, Bob deposits 1000 Atoms. Because the exchange rate has shifted, he received approximately 998 uAtoms, which are worth 1000 Atoms if he were to redeem them immediately.
+>
+> At week=3, Atom:uAtom increases to 1.003. Bob's 998 uAtom are now worth 1001 Atom if redeemed. Alice's 2000 uAtom are worth 2006 Atom.
+>
+> At week=4, Atom:uAtom increases to 1.004.
+>
+> Also at week=4, Alice redeems 1000 uAtom for 1004 Atom per the exchange rate.
+>
+> Also at week=4, Bob deposits an additional 1000 Atom, and receives 996 uAtom per the current exchange rate. His uAtom balance of 998+996 is worth 1002+1000 Atoms, as the two parts have grown by 0.2% and 0% respectively.
+
+This implementation sacrifices the "1:1 uToken to base asset exchange rate" and "uToken balances grow over time" facts promised in the whitepaper, while maintaining a mathematically identical incentive structure. In exchange, IBC transfer of uTokens becomes possible, interest transaction overhead is eliminated, and the ERC20 implementation becomes simplified.
+
+Specifically, because uTokens balances stored as ERC20 or as IBC voucher tokens do not need to grow in token amount, the question of how to "send new uTokens to all holders" disappears. A uToken's value increases no matter where it is held, by virtue of the Token:uToken exchange rate.
```

Comment:
```suggestion
Specifically, because uTokens balances stored as ERC20 or as IBC voucher tokens do not need to grow in token amount, the question of how to "send new uTokens to all holders" disappears.
A uToken's value increases no matter where it is held, by virtue of the Token:uToken exchange rate.
```